azurelinuxagent/ga/monitor.py

```diff
@@ -413,6 +413,10 @@ def init_cgroups():
             logger.verbose(traceback.format_exc())
 
     def send_cgroup_telemetry(self):
+        if not self.protocol.supports_telemetry:
```

Comment:
we don't want to disable this code; this also writes debug info to the local log, which would be essential when investigating resource usage issues.

Reply:
@narrieta  Makes sense.  I shifted this to short circuit only the call that breaks the abstraction.  If you'd like I am happy to push this up to be abstraction compliant I am happy to take a crack at it.  Just let me know